core/src/main/scala/org/apache/spark/util/logging/FileAppender.scala

```diff
@@ -76,7 +76,11 @@ private[spark] class FileAppender(inputStream: InputStream, file: File, bufferSi
           }
         }
       } {
-        closeFile()
+        try {
+          inputStream.close()
```

Comment:
I referred to `DriverRunner`:
DriverRunner-->runDriver-->redirectStream-->copyStream, the inputStream is closed in `copyStream`. @srowen